File: backend/src/Enum/ApplicationStatus.php
<?php

namespace App\Enum;

enum ApplicationStatus: string
{
    case DRAFT = 'DRAFT';
    case APPLIED = 'APPLIED';
    case WAITING = 'WAITING';
    case RELAUNCH = 'RELAUNCH';
    case INTERVIEW = 'INTERVIEW';
    case OFFER = 'OFFER';
    case REFUSED = 'REFUSED';
    case ABANDONED = 'ABANDONED';

    public function label(): string
    {
        return match($this) {
            self::DRAFT => 'À envoyer',
            self::APPLIED => 'Envoyée',
            self::WAITING => 'En attente',
            self::RELAUNCH => 'Relancé',
            self::INTERVIEW => 'Entretien',
            self::OFFER => 'Offre reçue',
            self::REFUSED => 'Refus',
            self::ABANDONED => 'Abandonné',
        };
    }

    public function isActive(): bool
    {
        return !in_array($this, [self::REFUSED, self::ABANDONED]);
    }
}

File: backend-java/src/main/java/com/jobdesk/domain/ApplicationStatus.java
package com.jobdesk.domain;

/**
 * Statut d'une candidature. Les constantes == valeurs stockées en base
 * (colonne varchar), donc mappé directement via @Enumerated(EnumType.STRING).
 */
public enum ApplicationStatus {
    DRAFT,
    APPLIED,
    WAITING,
    RELAUNCH,
    INTERVIEW,
    OFFER,
    REFUSED,
    ABANDONED
}
